timeseries/src/autogluon/timeseries/models/chronos/model.py

```diff
@@ -577,6 +570,15 @@ def _get_inference_data_loader(
         )
 
     def _get_context_length(self, data: TimeSeriesDataFrame) -> int:
+        # we validate the context_length here because the value is concrete,
+        # unlike in the constructor where it may be a search space
+        if self.context_length is not None and self.context_length > self.maximum_context_length:
+            logger.info(
+                f"\tContext length {self.context_length} exceeds maximum context length {self.maximum_context_length}."
+                f"Context length will be set to {self.maximum_context_length}."
+            )
+            self.context_length = self.maximum_context_length
+
```

Comment:
Should we maybe move all the `self.` attributes set during `__init__` (`batch_size`, `num_samples`, `device`, `torch_dtype`, `optimization_strategy`, `data_loader_num_workers` and `context_length`) to the `_fit` method? They all can potentially run into issues with HPO. 

We should try to use the access pattern with `model_params = self._get_model_params()` called only inside `_fit` (as in https://github.com/autogluon/autogluon/blob/master/timeseries/src/autogluon/timeseries/models/gluonts/abstract_gluonts.py#L235). These are guaranteed to contain "concrete" values, unlike the `hyperparameters` dict in `__init__` that may contain search spaces.

cc: @canerturkmen

Reply:
@abdulfatir, sorry, I only now noticed your comment above. Please ignore my previous message.